### PR TITLE
[cm]  4-point cubic hermite resampling

### PR DESCRIPTION
**Pull request by christhetree**
This PR adds a 4-point cubic hermite spline interpolation for better resampling.

The implementation is taken from "Polynomial Interpolators for High-Quality Resampling of Oversampled Audio" by Olli Niemitalo (http://yehar.com/blog/wp-content/uploads/2009/08/deip.pdf).
Does not dynamically allocate memory and is only ~2x slower than the `InplaceLinearResampler`. For comparison, sinc interpolation is ~4.5x slower and requires dynamic memory allocations.

The 2nd, -2nd, and sometimes -1st samples will be slightly off since this interpolator requires 4 points, but we assume a repeated sample when these are not available at the ends rather than adding 2-samples of delay. The 0th sample is always the same as the original audio since its corresponding x value is always 0.

From some experiments in Ableton, the spectrum shows about ~20 dB of improvement for a single sine tone going from 44.1 to 48 kHz at 2048 buffer size. From the frequency response of the hermite interpolator compared to the linear interpolater, we should see ~3 dB improvement in attenuation in the higher frequencies above 10 kHz.

44.1 kHz to 48 kHz (previous linear interpolator)
44_to_48_linear

44.1 kHz to 48 kHz (hermite interpolator)
44_to_48_hermite

88.2 kHz to 48 kHz (previous linear interpolator)
88_to_48_linear

88.2 kHz to 48 kHz (hermite interpolator)
88_to_48_hermite

cc: @Fyfe93

### CODE REVIEW

**Review by Fyfe93 — APPROVED** [review]
Code looks good and well documented. Couldn't see any issues with it.

File: neutone_sdk/sqw.py [hunk 1]
```diff
@@ -45,16 +46,15 @@ def __init__(
...
+        self.resample_sandwich = Inplace4pHermiteResampler(
             self.in_n_ch,
             self.out_n_ch,
             daw_sr,
             daw_sr,  # Tmp sample rate values
             daw_bs,
             use_debug_mode=use_debug_mode,
         )
-        self.params_resample_sandwich = InplaceInterpolationResampler(
+        self.params_resample_sandwich = InplaceLinearResampler(
```
Comment (bogdanteleaga) [inline neutone_sdk/sqw.py:57]: Is there some reason to keep this linear? Because it's faster and we don't care about accuracy in params as much?

**Review by bogdanteleaga — COMMENTED** [review]
LGTM, just left one comment.

File: neutone_sdk/sqw.py (hunk 1 above)
Comment (christhetree) [reply neutone_sdk/sqw.py:57]: Yeah the max frequency of the parameters is so much lower than audio (<100 Hz) that the signal is sufficiently oversampled such that a linear interpolator is actually close to optimal given the performance tradeoff.

**Pull request merged**